internal/controller/numaflowcontroller/numaflowcontroller_controller.go

```diff
@@ -546,16 +527,9 @@ func (r *NumaflowControllerReconciler) compareState(
 			IgnoreDifferences: sync.OverrideIgnoreDiff{JSONPointers: []string{"/status"}}},
 	}
 
-	resourceOps, cleanup, err := r.getResourceOperations()
-	if err != nil {
-		return gitopsSync.ReconciliationResult{}, nil, err
-	}
-	defer cleanup()
-
 	diffOpts := []diff.Option{
 		diff.WithLogr(*numaLogger.LogrLogger),
-		diff.WithServerSideDiff(true),
-		diff.WithServerSideDryRunner(diff.NewK8sServerSideDryRunner(resourceOps)),
+		diff.WithServerSideDiff(false),
```

Comment:
we don't need this (it just creates more opportunity for gitops-engine to error out)

Reply:
I believe Server Side Diff and Server Side Apply are used in the case that multiple parties may be modifying the same resource, which we don't need to deal with.